pkg/sdn/plugin/master.go

```diff
@@ -27,6 +27,7 @@ type OsdnMaster struct {
 
 func StartMaster(networkConfig osconfigapi.MasterNetworkConfig, osClient *osclient.Client, kClient *kclient.Client) error {
 	if !osapi.IsOpenShiftNetworkPlugin(networkConfig.NetworkPluginName) {
+		cleanupOpenShiftNetworkArtifacts(osClient)
```

Comment:
This is unfortunately not safe.  A misconfiguration on the master can remove all client info.  Instead, it would be more appropriate to log an event and an error to glog if any of the resources exist, and instruct administrators to run:

```
$ oc delete cluster-network,host-subnets,net-namespaces --all
```